fix(files_sharing): Do not wrap password policy exception into a generic one

come-nc opened the pull request:
Follow-up of https://github.com/nextcloud/server/pull/49361

## Summary

Let the controller access the HintException and show the error to the user.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

come-nc committed:
fix(files_sharing): Do not wrap password policy exception into a generic one

Let the controller access the HintException and show the error to the user.

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

susnux commented on lib/private/Share20/Manager.php:
```diff
@@ -111,11 +111,7 @@ protected function verifyPassword($password) {
-		} catch (HintException $e) {
-			throw new \Exception($e->getHint());
-		}
+		$this->dispatcher->dispatchTyped(new ValidatePasswordPolicyEvent($password));
```
Still disappointed that this is done using exceptions rather than setting property on the even like `isValid`...

tobiasKaminsky reviewed (changes requested):
Unfortunately this does not fix it

come-nc commented:
> Unfortunately this does not fix it

I’m out of ideas

come-nc commented:
> Do you need more info how to reproduce it? Me as Android dev cannot really help, expect of steps to reproduce…

Do you have a full stacktrace for the Exception? (with this PR applied on server side)

come-nc committed:
fix(shares): Wrap exceptions from password validation to set code to 400

This fixes a regression that bad password returned 403 instead of 400
 because of previous changes.

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

come-nc committed:
fix(tests): Adapt ManagerTest to change in Exception class used

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

come-nc commented:
@tobiasKaminsky Can you test again? It should be good now, message should be back and error code should be 400.

tobiasKaminsky commented:
This works now without any modification on client side :tada:

tobiasKaminsky commented:
@come-nc can we merge this? :blue_heart:
